Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -168,8 +168,28 @@ extension ProjectDescription.Project {
         productToPackage: [String: String],
         targetDependencyToFramework: [String: Path]
     ) throws -> Self {
-        let targets = try packageInfo.targets.compactMap { target in
-            try Target.from(
+        var relevantTargets: [String] = []
+        var targetsToProcess = packageInfo.products.flatMap { $0.targets }.uniqued()
+        while !targetsToProcess.isEmpty {
```

Comment:
This mapping function is gaining complexity that I think it's worth extracting into its own method that we can test. What about adding a function to `PackageInfo`?

```swift
struct PackageInfo {
  func productTargetDependencies() -> [WhateverType]
}
```
Note that I renamed **relevant** to **productTargetDependencies** that I think captures better what the targets are. 

Moreover, we are traversing a graph without keeping track of the nodes we've already gone through. This might lead to infinite loop processing targets. I think you might need to add some dictionary that keeps track of those nodes, and avoid passing through them again if you've already analyzed them.

Reply:
makes sense, I'll do a separate PR starting after the current last open PR to avoid too much conflicts, makes sense? 🙏